packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -16,33 +16,81 @@
 
 import { LogLevel } from '../common/types';
 
-export type ENVIRONMENT_MAP = { [key: string]: string | number };
+const DEFAULT_LIST_SEPARATOR = ',';
 
 /**
  * Environment interface to define all names
  */
-export interface ENVIRONMENT {
-  OTEL_LOG_LEVEL?: LogLevel;
-  OTEL_NO_PATCH_MODULES?: string;
+
+type ENVIRONMENT_NUMBERS = {
   OTEL_SAMPLING_PROBABILITY?: number;
   OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT?: number;
   OTEL_SPAN_EVENT_COUNT_LIMIT?: number;
   OTEL_SPAN_LINK_COUNT_LIMIT?: number;
-}
+};
 
-const ENVIRONMENT_NUMBERS: Partial<keyof ENVIRONMENT>[] = [
+const ENVIRONMENT_NUMBERS_KEYS: Partial<keyof ENVIRONMENT_NUMBERS>[] = [
   'OTEL_SAMPLING_PROBABILITY',
   'OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT',
   'OTEL_SPAN_EVENT_COUNT_LIMIT',
   'OTEL_SPAN_LINK_COUNT_LIMIT',
 ];
 
+function isEnvVarANumber(key: unknown): key is keyof ENVIRONMENT_NUMBERS {
+  return (
+    ENVIRONMENT_NUMBERS_KEYS.indexOf(key as keyof ENVIRONMENT_NUMBERS) > -1
+  );
+}
+
+type ENVIRONMENT_LISTS = {
+  OTEL_NO_PATCH_MODULES?: string[];
+};
+
+const ENVIRONMENT_LISTS_KEYS: Partial<keyof ENVIRONMENT_LISTS>[] = [
+  'OTEL_NO_PATCH_MODULES',
+];
+
+function isEnvVarAList(key: unknown): key is keyof ENVIRONMENT_LISTS {
+  return ENVIRONMENT_LISTS_KEYS.indexOf(key as keyof ENVIRONMENT_LISTS) > -1;
+}
+
+export type ENVIRONMENT = {
+  CONTAINER_NAME?: string;
+  ECS_CONTAINER_METADATA_URI_V4?: string;
+  ECS_CONTAINER_METADATA_URI?: string;
+  HOSTNAME?: string;
+  KUBERNETES_SERVICE_HOST?: string;
+  NAMESPACE?: string;
+  OTEL_EXPORTER_JAEGER_AGENT_HOST?: string;
+  OTEL_EXPORTER_JAEGER_ENDPOINT?: string;
+  OTEL_EXPORTER_JAEGER_PASSWORD?: string;
+  OTEL_EXPORTER_JAEGER_USER?: string;
+  OTEL_LOG_LEVEL?: LogLevel;
+  OTEL_RESOURCE_ATTRIBUTES?: string;
+} & ENVIRONMENT_NUMBERS &
+  ENVIRONMENT_LISTS;
+
+export type RAW_ENVIRONMENT = {
```

Comment:
hmm why changing the name of that (ENVIRONMENT_MAP) ?

Reply:
You have asked a couple of questions around this area, let's discuss them all here. My thoughts:
- names of types for environment: raw data from `process.env` and extracted mapped data. Now I suggested a name different than `ENVIRONMENT_MAP`, because "map" is very close in meaning to a hashmap/object, which is what `process.env` already is, hence `RAW_ENVIRONMENT` and `ENVIRONMENT`, but let me know what you think
- I changed the type on the returned variable `environment`, to be the same as the return type on the function, I don't quite understand why they were different before